### PR TITLE
Operator Spec PackageName

### PR DESCRIPTION
**Pull request by dtfranz**
Addresses Issue #69 
Add package name to spec with validation, clear generated comments, run make, update sample CR

Tested by running:
```
kind create cluster
make install
kubectl apply -f config/samples/operators_v1alpha1_operator.yaml
```

### CODE REVIEW

File: config/samples/operators_v1alpha1_operator.yaml [hunk 1]
```diff
@@ -9,4 +9,4 @@ metadata:
     app.kubernetes.io/created-by: operator-controller
   name: operator-sample
 spec:
-  # TODO(user): Add fields here
```
Comment (tmshort) [inline config/samples/operators_v1alpha1_operator.yaml:12]: Keep the TODO comment for future field adders?

File: config/crd/bases/operators.operatorframework.io_operators.yaml [hunk 2]
```diff
@@ -35,10 +35,12 @@ spec:
           spec:
             description: OperatorSpec defines the desired state of Operator
             properties:
-              foo:
-                description: Foo is an example field of Operator. Edit operator_types.go
-                  to remove/update
+              packageName:
+                maxLength: 48
+                pattern: ^[a-z0-9]+(-[a-z0-9]+)*$
```
Comment (tmshort) [inline config/crd/bases/operators.operatorframework.io_operators.yaml:40]: Rules out `camelCase`, but I did see the discussion that this matches all current catalogs.

Comment (dtfranz) [reply config/crd/bases/operators.operatorframework.io_operators.yaml:40]: True, but I think there's a good chance we'll be generating cluster resources with these names, and if that's the case we'd need to restrict camel case anyway

File: config/samples/operators_v1alpha1_operator.yaml (hunk 1 above)
Comment (dtfranz) [reply config/samples/operators_v1alpha1_operator.yaml:12]: Sure!

Comment (dtfranz) [reply config/samples/operators_v1alpha1_operator.yaml:12]: Updated

**Review by tmshort — APPROVED** [review]
/lgtm

File: api/v1alpha1/operator_types.go [hunk 3]
```diff
@@ -20,23 +20,15 @@ import (
...
-
 // OperatorSpec defines the desired state of Operator
 type OperatorSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
-
-	// Foo is an example field of Operator. Edit operator_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
+	//+kubebuilder:validation:MaxLength:=48
+	//+kubebuilder:validation:Pattern:=^[a-z0-9]+(-[a-z0-9]+)*$
```
[the remark this answers is not in the document]
Comment (awgreene) [reply api/v1alpha1/operator_types.go:26]: hmm, 3scale operator jumps to mind.

[the remark this answers is not in the document]
Comment (dtfranz) [reply api/v1alpha1/operator_types.go:26]: Yup, that's exactly why I allowed the numeral in there

**Comment by tmshort** [comment]
Any other approval needed? @awgreene ?

**Comment by tmshort** [comment]
Merged. Thank you!

**Pull request merged**